rsocket-transport-netty/src/test/java/io/rsocket/integration/KeepaliveTest.java

```diff
@@ -0,0 +1,188 @@
+
+    TcpServer tcpServer = TcpServer.create().host("localhost").port(PORT);
+
+    return RSocketServer.create(
+            (setupPayload, rSocket) -> {
+              rSocket
+                  .onClose()
+                  .doFirst(() -> LOG.info("Connected on server side."))
+                  .doOnTerminate(() -> LOG.info("Connection closed on server side."))
+                  .subscribe();
+
+              return Mono.just(new MyServerRsocket());
+            })
+        .payloadDecoder(PayloadDecoder.ZERO_COPY)
+        .bind(TcpServerTransport.create(tcpServer))
+        .doOnNext(closeableChannel -> LOG.info("RSocket server started."));
+  }
+
+  private static RSocketClient createClient() {
+    LOG.info("Connecting....");
+
+    Function<String, RetryBackoffSpec> reconnectSpec =
+        reason ->
+            Retry.backoff(Long.MAX_VALUE, Duration.ofSeconds(10L))
+                .doBeforeRetry(retrySignal -> LOG.info("Reconnecting. Reason: {}", reason));
+
+    Mono<RSocket> rsocketMono =
+        RSocketConnector.create()
+            .fragment(16384)
+            .reconnect(reconnectSpec.apply("connector-close"))
+            .keepAlive(Duration.ofMillis(100L), Duration.ofMillis(900L))
+            .connect(TcpClientTransport.create(TcpClient.create().host("localhost").port(PORT)));
+
+    RSocketClient client = RSocketClient.from(rsocketMono);
+
+    client
+        .source()
+        .doOnNext(r -> LOG.info("Got RSocket"))
+        .flatMap(RSocket::onClose)
+        .doOnError(err -> LOG.error("Error during onClose.", err))
+        .retryWhen(reconnectSpec.apply("client-close"))
+        .doFirst(() -> LOG.info("Connected on client side."))
+        .doOnTerminate(() -> LOG.info("Connection closed on client side."))
+        .repeat()
+        .subscribe();
+
+    return client;
+  }
+
+  private static Mono<RSocket> createClientLazy() {
+    LOG.info("Connecting....");
+
+    Function<String, RetryBackoffSpec> reconnectSpec =
+        reason ->
+            Retry.backoff(Long.MAX_VALUE, Duration.ofSeconds(10L))
+                .doBeforeRetry(retrySignal -> LOG.info("Reconnecting. Reason: {}", reason));
+
+    return RSocketConnector.create()
+        .fragment(16384)
+        .reconnect(reconnectSpec.apply("connector-close"))
+        .keepAlive(Duration.ofMillis(100L), Duration.ofMillis(900L))
+        .connect(TcpClientTransport.create(TcpClient.create().host("localhost").port(PORT)));
+
+    //		RSocketClient client = RSocketClient.from(rsocketMono);
+
+    //		client
+    //				.source()
+    //				.doOnNext(r -> LOG.info("Got RSocket"))
+    //				.flatMap(RSocket::onClose)
+    //				.doOnError(err -> LOG.error("Error during onClose.", err))
+    //				.retryWhen(reconnectSpec.apply("client-close"))
+    //				.doFirst(() -> LOG.info("Connected on client side."))
+    //				.doOnTerminate(() -> LOG.info("Connection closed on client side."))
+    //				.repeat()
+    //				.subscribe();
+
+    //		return client;
+  }
```

Comment:
The createClient methods maybe can also be combined into one with a `lazy` input argument.